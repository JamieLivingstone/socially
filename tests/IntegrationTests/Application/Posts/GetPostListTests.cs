using System.Threading.Tasks;
using Application.Common.Exceptions;
using Application.Posts.Queries.GetPostList;
using FluentAssertions;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Posts;

[TestFixture]
public class GetPostListTests : TestBase
{
  [Test]
  public void GivenAnInvalidRequest_ThrowsValidationException()
  {
    var query = new GetPostListQuery
    {
      PageNumber = 0, // Must be 1 or greater
      PageSize = 10
    };

    FluentActions.Invoking(() => SendAsync(query)).Should().ThrowAsync<ValidationException>();
  }

  [Test]
  public async Task GivenFiltersHasNoCorrespondingPosts_ReturnsEmptyList()
  {
    var query = new GetPostListQuery
    {
      PageNumber = 1,
      PageSize = 1,
      Author = "does_not_exist"
    };

    var result = await SendAsync(query);

    Snapshot.Match(result);
  }

  [Test]
  public async Task GivenAPaginatedRequestWithCustomOrdering_ReturnsPostList()
  {
    var query = new GetPostListQuery
    {
      PageNumber = 2,
      PageSize = 1,
      OrderBy = PostListOrder.Created
    };

    var result = await SendAsync(query);

    Snapshot.Match(result);
  }
}
