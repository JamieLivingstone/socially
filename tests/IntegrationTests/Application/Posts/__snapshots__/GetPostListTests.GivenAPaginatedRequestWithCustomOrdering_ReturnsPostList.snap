{
  "Items": [
    {
      "Slug": "test-post-two",
      "Title": "Test Post Two",
      "CreatedAt": "2020-05-04T00:00:00",
      "Author": {
        "Name": "Peter Gray",
        "Username": "peter"
      },
      "Tags": [],
      "CommentsCount": 0,
      "LikesCount": 0
    }
  ],
  "PageNumber": 2,
  "TotalPages": 2,
  "TotalCount": 2,
  "HasPreviousPage": true,
  "HasNextPage": false
}
